addons/web/static/src/js/core/dialog.js

```diff
@@ -155,7 +166,20 @@ var Dialog = Widget.extend({
         this.appendTo($('<div/>')).then(function () {
             self.$modal.find(".modal-body").replaceWith(self.$el);
             self.$modal.modal('show');
-            self._opened.resolve();
+            self._opened.resolve().then(function () {
```

Comment:
It would be way better if the dialog was opened with the correct size rather than modifying it once it has been opened. I saw that you were using a *lot* of less to style this dialog so maybe it can be done with less ? If it is not and you need more expressive power, you can even create a new FullScreenDialog class that does exactly this. I think both cases would be better than modifying the css of the modal after it was already opened.

Reply:
Actually in Full-Screen modal, we need header and footer of the modal to be 'fixed', and modal-body to be scrollable. Now the problem is, we can't give fixed height to the footer as it may vary depending on number of buttons it has(and even on language / Caption of the button). For example, footer with four buttons will have nearly double height than the footer with two buttons. In that case, modal-body should be up-lifted accordingly to make sure no content is hidden behind the sticky footer. And I could't find any easy way to do this using css.

Reply:
Ok but why couldn't you run this js code inside the FullScreenDialog class right before the modal is actually opened rather than right after ?

Reply:
Because, we can't know the size of header/footer before it is opened.